spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/DefaultZipkinRestTemplateCustomizer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.zipkin;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Default {@link ZipkinRestTemplateCustomizer} that can customize interceptor
+ * to the {@link RestTemplate} responsible for sending spans to Zipkin.
+ *
+ * @author Marcin Grzejszczak
+ *
+ * @since 1.1.0
+ */
+public class DefaultZipkinRestTemplateCustomizer implements ZipkinRestTemplateCustomizer {
+	private final ZipkinProperties zipkinProperties;
+
+	public DefaultZipkinRestTemplateCustomizer(
+			ZipkinProperties zipkinProperties) {
+		this.zipkinProperties = zipkinProperties;
+	}
+
+	@Override
+	public void customize(RestTemplate restTemplate) {
+		if (this.zipkinProperties.getCompression().isEnabled()) {
```

Comment:
Hmm I don't think I agree here. From the code's perspective (the `HttpZipkinSpanReporter`) it's just sending spans. Whether they are compressed or not - he shouldn't care. It's a technical decision of the person configuring the `RestTemplate` so the Http Client. At the moment the default customization of the HTTP client is that it creates a compression mechanism if there is a property. But someone can override the RestTemplate with any sort of an approach he wants to. 

I can agree that we could create a  `RestTemplate` that would by default have the compression mechanism turned on and then the customizer can disable it. But I'd like to stick with the separation of concerns.